finagle-backend/src/main/scala/sttp/client/finagle/FinagleBackend.scala

```diff
@@ -0,0 +1,165 @@
+package sttp.client.finagle
+
+import com.twitter.finagle.Http.Client
+import com.twitter.finagle.{Http, http}
+import sttp.client.{BasicRequestBody, ByteArrayBody, ByteBufferBody, FileBody, FollowRedirectsBackend, IgnoreResponse, InputStreamBody, MappedResponseAs, MultipartBody, NoBody, NothingT, Request, Response, ResponseAs, ResponseAsByteArray, ResponseAsFile, ResponseAsFromMetadata, ResponseAsStream, ResponseMetadata, StringBody, SttpBackend}
+import com.twitter.util.{Try, Future => TFuture}
+import sttp.client.monad.MonadError
+import sttp.client.ws.WebSocketResponse
+import com.twitter.finagle.http.{FileElement, FormElement, RequestBuilder, SimpleElement, Method => FMethod, Response => FResponse}
+import com.twitter.io.Buf
+import com.twitter.io.Buf.{ByteArray, ByteBuffer}
+import com.twitter.util
+import sttp.client.internal.FileHelpers
+import sttp.model.{Header, Method, Part, StatusCode}
+import com.twitter.finagle.util.DefaultTimer
+import com.twitter.util.Duration
+import scala.io.Source
+
+class FinagleBackend(client: Client) extends SttpBackend[TFuture, Nothing, NothingT] {
+  override def send[T](request: Request[T, Nothing]): TFuture[Response[T]] = {
+    val service = client.newService(s"${request.uri.host}:${request.uri.port.getOrElse(80)}")
+    val finagleRequest = requestBodyToFinagle(request)
+    service.apply(finagleRequest).within(Duration.fromMilliseconds(request.options.readTimeout.toMillis))(DefaultTimer).flatMap{
```

Comment:
I suspect `within` will complete the future in case of a timeout - but maybe finagle has some built-in timeout methods? So that the connection can be closed (and not only the future completed, with the other branch still running)

Reply:
Finagle does have a built in [request timeout](https://twitter.github.io/finagle/docs/com/twitter/finagle/Http$Client.html#withRequestTimeout(timeout:com.twitter.util.Duration):com.twitter.finagle.Http.Client ) , will use that